src/bin/main.rs

```diff
@@ -159,34 +185,56 @@ fn main() {
         .map(|ep| ep)
         .or_else(|| paths.entrypoint.as_deref());
 
-    let inputs = matches.values_of("input").map(|files| {
-        PathUtil::list_all_files_str(&files.into_iter().collect::<Vec<_>>()[..])
-            .unwrap()
-            .into_iter()
-            .map(|file| file.clone())
-            .collect()
-    });
+    let input_patterns: Option<Vec<_>> = matches
+        .values_of("input")
+        .map(|patterns| patterns.map(|pattern| pattern.to_string()).collect())
+        .or_else(|| paths.files.to_owned());
 
-    let inputs: Vec<_> = match inputs.or_else(|| {
-        paths.files.as_ref().map(|files| {
-            PathUtil::list_all_files(&files[..])
-                .unwrap()
-                .into_iter()
-                .map(|file| file.clone())
-                .collect()
-        })
-    }) {
-        Some(inputs) => inputs,
+    let input_patterns = match input_patterns {
         None => {
-            eprintln!(
-                "No inputs provided via arguments or toml file. For help, use:\n\
+            return Err(format!(
+                "ERROR: No inputs provided via arguments or toml file. For help, use:\n\
                  > yarner -h",
-            );
-            return;
+            ))
         }
+        Some(patterns) => patterns,
     };
 
-    for input in inputs {
+    let mut inputs: Vec<PathBuf> = Vec::new();
+    for pattern in input_patterns {
```

Comment:
Due to the "deep boxing" of results in vectors in results in option (or similar) I was not able to solve this using comprehensions, so it does not look that "rusty". However, I got rid of `PathUtil`.

Reply:
Moved file procesing into that loop, so no storage for inputs in vector anymore.